instagram/src/authentication/withAuthenticate.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+
+const withAuthenticate = FirstComponent => SecondComponent => props => 
+    class extends React.Component {
+        constructor(props){
+            super(props);
+            this.state= {
+                loggedIn: "true"
+            }
+        }
+        render (){
+            // if (loggedIn) {
```

Comment:
```suggestion
            // if (this.state.loggedIn) {
```
you have to make sure you reference the state